app/jobs/create_rss_notifications.rb

```diff
@@ -5,7 +5,7 @@ class CreateRssNotifications < ApplicationJob
 
   def perform(options = {})
     # Defaults to theforeman.org blog RSS
-    UINotifications::RssNotificationsChecker.new(options).deliver!
+    UINotifications::RssNotificationsChecker.new(options || {}).deliver!
```

Comment:
while this does not hurt, I still don't understand, why this would be required. This could only happen if the method is called with `nil` as an argument. The cause seems to be at https://github.com/theforeman/foreman/blob/develop/app/jobs/create_rss_notifications.rb#L3 the `arguments.first` could cause it, I suppose we should simply `perform_later` with `job.arguments`

Reply:
My issue with it is that although the `first` method is driven from `Enumerable`, it is more like an `Array` or `Set` (for example) related, and less good for a `Hash`. With `Hash` you don't really know the order of the key/value pairs, and it will break it into `{key, value}` result.

So it means that `job.arguments` is an array in this case, just from the syntax, before checking the documentation about it (that I could not find btw).

The thing is, is that doing `job.arguments[:foo]` for example, will result with the following exception:
```
TypeError: no implicit conversion of Symbol into Integer
```

And as we know based on the stack, it will arrive to a method that looks for a Hash rather then an array.

So checking for `nil` is better imo, then passing an `Array`, what do you think?

Reply:
> My issue with it is that although the first method is driven from Enumerable, it is more like an Array or Set (for example) related, and less good for a Hash. With Hash you don't really know the order of the key/value pairs, and it will break it into {key, value} result.

absolutely, the issue IMHO is the fact, that we do `job.arguments.first`, we should simply pass job.arguments, which are expected to be hash based on what's here [perform method](https://github.com/theforeman/foreman/blob/develop/app/jobs/create_rss_notifications.rb#L6)

Reply:
Now I get it, sorry, added the fix.

Reply:
no worries, I guess this change is not required anymore, `options` should now always be a hash so I'd recommend removing `|| {}` from line 8, then it's good to be merged

Reply:
done :)